library/network/src/modules/SuSEFirewall.rb

```diff
@@ -1249,9 +1249,75 @@ def GetProtectFromInternalZone
     # @example GetInterfacesInZone ("external") -> ["eth4", "eth5"]
     def GetInterfacesInZone(zone)
       return [] if !GetKnownFirewallZones().include?(zone)
-      deep_copy(@SETTINGS[zone][:interfaces])
+      # Only return the known ones
+
+      interfaces_in_zone = deep_copy(@SETTINGS[zone][:interfaces])
+
+      known_interfaces_now = GetListOfKnownInterfaces()
+
+      # filtering special strings
+      interfaces_in_zone = Builtins.filter(interfaces_in_zone) do |interface|
+        interface != "" && Builtins.contains(known_interfaces_now, interface)
```

Comment:
The "" comparison is a relict from the old method which had to guard against multiple spaces in a `splitstring` result.
So this can become
```rb
return [] unless IsKnownZone(zone)
known_interfaces_now = GetListOfKnownInterfaces()
@SETTINGS[zone][:interfaces].find_all {|i| known_interfaces_now.include?(i) }
```